stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -536,25 +546,41 @@ extension TFState {
         debugLog("Running enqueue with \(inputTensors.count) input tensors.")
       }
     }
+    var runOptions: UnsafePointer<TF_Buffer>? = nil
+    var runMetadataOutput: UnsafeMutablePointer<TF_Buffer>? = nil
+    // When there's a run metadata output path specified, we enable `FULL_TRACE`
+    // in run options and dump that to the specified path after execution.
+    if _RuntimeConfig.runMetadataOutputPath != nil {
+      runOptions = UnsafePointer(TF_CreateRunOptions(/*enable_full_trace*/ 1))
+      runMetadataOutput = TF_NewBuffer()
+    }
     debugLog("""
-               Calling TF_SessionRun on function \(entryFunctionBaseName), With \
-               \(targetNodeSpecs.count) target nodes.
-               """)
+      Calling TF_SessionRun on function \(entryFunctionBaseName), With \
+      \(targetNodeSpecs.count) target nodes.
+      """)
     TF_SessionRun(
-      cSession, nil,
+      cSession, runOptions,
       // input related parameters
       inputNodeSpecs, inputTensors, Int32(inputTensors.count),
       // output related parameters
       outputNodeSpecs, &outputTensors, Int32(returnValues.count),
       // target related parameters
       targetNodeSpecs, Int32(targetNodeSpecs.count),
-      /*run_metadata*/nil, status
+      /*run_metadata*/ runMetadataOutput, status
     )
     if (TF_GetCode(status) != TF_OK) {
       _ = fputs(TF_Message(status), stderr)
       exit(-1)
     }
     debugLog("Done running TF computation.")
+    
+    // If run metadata path was set, dump the run metadata proto to a file.
+    if let path = _RuntimeConfig.runMetadataOutputPath {
+      TF_DeleteBuffer(runOptions)
+      writeContents(of: runMetadataOutput!, toFile: path)
+      TF_DeleteBuffer(runMetadataOutput)
+      debugLog("Writing run metadata to \"\(path)\"")
```

Comment:
nit: consider logging this before the writeContents() call.

Reply:
Done.